src/main/java/org/eclipse/xpanse/terraform/boot/security/zitadel/config/ZitadelWebSecurityConfig.java

```diff
@@ -0,0 +1,214 @@
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtDecoders;
+import org.springframework.security.oauth2.jwt.JwtIssuerValidator;
+import org.springframework.security.oauth2.jwt.JwtTimestampValidator;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.header.writers.frameoptions.XFrameOptionsHeaderWriter;
+import org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.web.cors.CorsConfiguration;
+import org.springframework.web.cors.CorsConfigurationSource;
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
+import org.springframework.web.servlet.handler.HandlerMappingIntrospector;
+
+/**
+ * Configuration applied on all web endpoints defined for this
+ * application. Any configuration on specific resources is applied
+ * in addition to these global rules.
+ */
+@Slf4j
+@Profile("zitadel")
+@Configuration
+@EnableWebSecurity
+@EnableMethodSecurity(securedEnabled = true)
+public class ZitadelWebSecurityConfig {
+
+    @Value("${authorization-token-type:JWT}")
+    private String authTokenType;
+
+    @Value("${spring.security.oauth2.resourceserver.jwt.issuer-uri}")
+    private String issuerUri;
+
+    @Value("${spring.security.oauth2.resourceserver.opaquetoken.introspection-uri}")
+    private String introspectionUri;
+
+    @Value("${spring.security.oauth2.resourceserver.opaquetoken.client-id}")
+    private String clientId;
+
+    @Value("${spring.security.oauth2.resourceserver.opaquetoken.client-secret}")
+    private String clientSecret;
+
+    /**
+     * Configures basic security handler per HTTP session.
+     *
+     * @param http security configuration
+     */
+    @Bean
+    public SecurityFilterChain apiFilterChain(HttpSecurity http,
+                                              HandlerMappingIntrospector introspector)
+            throws Exception {
+        // accept cors requests and allow preflight checks
+        http.cors(httpSecurityCorsConfigurer -> httpSecurityCorsConfigurer.configurationSource(
+                corsConfigurationSource()));
+
+        MvcRequestMatcher.Builder mvcMatcherBuilder =
+                new MvcRequestMatcher.Builder(introspector).servletPath("/");
+
+        http.authorizeHttpRequests(arc -> {
+            arc.requestMatchers(AntPathRequestMatcher.antMatcher("/h2-console/**")).permitAll();
```

Comment:
route does not exist. must be removed. Please check all other routes too. Many are not valid.